magicctapipe/scripts/lst1_magic/semi_automatic_scripts/DL2_to_DL3.py

```diff
@@ -0,0 +1,338 @@
+    """
+    if cluster != "SLURM":
+        logger.warning(
+            "Automatic processing not implemented for the cluster indicated in the config file"
+        )
+        return
+
+    # Loop over all nights
+
+    DL3_Nights = np.sort(
+        glob.glob(f"{target_dir}/v{__version__}/{source}/DL3/{IRF_cuts_type}/*")
+    )
+    for dl3date in DL3_Nights:
+        night = dl3date.split("/")[-1]
+        outdir = f"{dl3date}/logs"
+        File_list = np.sort(glob.glob(f"{outdir}/*.txt"))
+        for file in File_list:
+
+            if str(night) not in LST_date:
+                continue
+            with open(file, "r") as f:
+                runs = f.readlines()
+                run_new = []
+                for run in runs:
+                    run_number = run.split("/")[-1].split(".")[-2].split("n")[-1]
+                    wobble_offset = df_LST[df_LST.LST1_run == run_number].iloc[0][
+                        "wobble_offset"
+                    ]
+                    if str(wobble_offset) != "['0.40']":
+                        print(
+                            f"wobble offset is not (or not always) 0.40 for {source}, run {run_number}"
+                        )
+                        continue
+                    single_run_new = (
+                        "/".join(run.split("/")[:-6])
+                        + f"/v{version}/"
+                        + run.split("/")[-5]
+                        + "/DL2/"
+                        + run.split("/")[-2]
+                        + "/"
+                        + run.split("/")[-1].replace("dl1_stereo", "dl2")
+                    )
+                    run_new.append(single_run_new)
+                process_size = len(run_new) - 1
+                with open(file, "w") as g:
+                    g.writelines(run_new)
+
+            nsb = file.split("/")[-1].split("_")[1]
+            period = file.split("/")[-1].split("_")[0]
+            if process_size < 0:
+                print(
+                    f"No runs to be processed for {source}, NSB {nsb}, period {period}"
+                )
+                continue
+            dec = df_LST[df_LST.source == source].iloc[0]["MC_dec"]
+            if np.isnan(dec):
+                print(f"MC_dec is NaN for {source}")
+                continue
+            dec = str(dec).replace(".", "").replace("-", "min_")
+            IRFdir = f"{IRF_dir}/{period}/NSB{nsb}/GammaTest/v{MC_v}/{IRF_cuts_type}/dec_{dec}{'_high_density' if source in dense_list else ''}/"
+            if (not os.path.isdir(IRFdir)) or (
+                len(glob.glob(f"{IRFdir}/irf_*fits.gz")) < 1
+            ):
+                print(f"no IRF availables in {IRFdir}")
+                continue
+
+            slurm = slurm_lines(
+                queue="short",
+                job_name=f"{source}_DL2_to_DL3",
+                nice_parameter=nice,
+                array=process_size,
+                mem="50g",
```

Comment:
do you really need 50 GB? the IRF files are tiny, and DL2 files are not that huge, a guess a few GB should suffice. I was normally running with 3GB, but you could add a bit of a margin to this to be on a safe side